pkg/integrationteststatus/integration_test_status.go

```diff
@@ -168,10 +172,10 @@ func (sits *SnapshotIntegrationTestStatuses) UpdateTestStatusIfChanged(scenarioN
 func (sits *SnapshotIntegrationTestStatuses) UpdateTestPipelineRunName(scenarioName string, pipelineRunName string) error {
 	detail, ok := sits.GetScenarioStatus(scenarioName)
 	if !ok {
-		return fmt.Errorf("scenario name %s not found and cannot be updated", scenarioName)
+		return fmt.Errorf("scenario name %s not found within the SnapshotIntegrationTestStatus, and cannot be updated", scenarioName)
 	}
 
-	if detail.TestPipelineRunName != pipelineRunName {
+	if detail.TestPipelineRunName != pipelineRunName && detail.Status != IntegrationTestStatusCancelled {
```

Comment:
Why is this needed? Even cancelled pipeline has still name

Reply:
but if we add the name of a canceled PLR, it won't retrigger the creation of a new PLR [here](https://github.com/redhat-appstudio/integration-service/blob/2880904d7988527ce384ad14a5f5c0ece9c06e97/controllers/snapshot/snapshot_adapter.go#L127)

Reply:
alternatively, I can update the if statement within the snapshot_adapter:
https://github.com/redhat-appstudio/integration-service/blob/2880904d7988527ce384ad14a5f5c0ece9c06e97/controllers/snapshot/snapshot_adapter.go#L127

to this:

```
if ok && integrationTestScenarioStatus.TestPipelineRunName != "" && integrationTestScenarioStatus.Status != intgteststat.IntegrationTestStatusCancelled
```

wdyt?

Reply:
How exactly it would help? Status would have already recorded pipeline name from InProgress status.

Do we even want to restart cancelled pipeline?

Reply:
If you want to restart pipeline, it must be handled [here](https://github.com/redhat-appstudio/integration-service/blob/2880904d7988527ce384ad14a5f5c0ece9c06e97/controllers/snapshot/snapshot_adapter.go#L127)
```
if pipeline is cancelled:
    recreate pipeline
```

Reply:
I was wrong, it makes sense to remove it, but we shouldn't call status cancelled, because cancelled is something else. Cancelled pipeline is still valid pipeline that's kept in k8s.

Deleted would be more specific

https://tekton.dev/docs/pipelines/pipelineruns/#pipelinerun-status

Reply:
Tbh I'm not sure if we do want to restart cancelled pipeline runs - there are definitely scenarios where we wouldn't want to do that automatically, e.g. when the test uses an ephemeral environment that costs money/resources and the user cancelled it because of an issue in testing where it got stuck.

In any case I don't think that restarting cancelled integration pipelines should be handled as part of this ticket.

Reply:
not sure if I understand how it is different from what I proposed above.

> Status would have already recorded pipeline name from InProgress status.

Yes, and that's why with the addition of an extra condition to ensure status is not "Cancelled" (described above), the controller would create a new PLR, am I missing something?

Reply:
I had still old PR before force push in memory and it was not there, so my comments are off now, I had to start all over again.

Reply:
> I was wrong, it makes sense to remove it

Taking back what I said, it should keep pipelinerun name, it express the status that pipeline with this particular name has been deleted.

Reply:
Thanks guys for the explanation.

Do we then eventually, expect to see this when a user deletes a PLR:
* Mark the snapshot as "Failed", and also update its annotation ofcourse
* remove the finalizer if PLR was not created by a PR event -> "Deleted" PLR gets deleted.

Reply:
Yes, that sounds correct to me.

Reply:
Perfect

Reply:
Updated the logic to reflect the conclusion of this discussion.

Now when an Intg PLR is deleted while it is running, its status within the Snapshot's status annotation gets updated to "Deleted", and Snapshot gets marked as Failed and Finished, and the Finalizer is removed from the PLR.